servo/connectors/kubernetes.py

```diff
@@ -3564,6 +3564,23 @@ async def attach(self, servo_: servo.Servo) -> None:
         # Ensure we are ready to talk to Kubernetes API
         await self.config.load_kubeconfig()
 
+        # Set K8s version and target namespace for user agent
+        with self.logger.catch(level="DEBUG", message=f"Unable to set User Agent string for connector {self.name}"):
+            version = "UnknownVersion"
+            comment_dict = { "namespace": self.config.namespace }
+            async with kubernetes_asyncio.client.api_client.ApiClient() as api:
+                v1 =kubernetes_asyncio.client.VersionApi(api)
+                version_obj = await v1.get_code()
+                version = f"{version_obj.major}.{version_obj.minor}"
+                comment_dict["platform"] = version_obj.platform
+
+            comment_str = "; ".join(list(map(lambda tup: f"{tup[0]} {tup[1]}", comment_dict.items())))
+            self.optimizer.add_user_agent(self.name, f"{version} ({comment_str})")
```

Comment:
Maybe we use the shared config cascading stuff to expose `environment` or `telemetry` so that all connectors can do `self.telemetry['kubernetes.version'] = ...`

We'll also want to get some data from other connectors like the Prometheus like the Prometheus version we are talkiing to.